priv/templates/message.ex.eex

```diff
@@ -9,6 +9,14 @@ defmodule <%= name %> do
   def full_name do
     "<%= full_name %>"
   end
+  <%= if not is_nil(message_options) do %>
+  def message_options do
+    # credo:disable-for-next-line
+    <%= message_options.__struct__ %>.decode(
+      <%= message_options.__struct__.encode(message_options |> Map.from_struct() |> Enum.filter(fn {_, x} -> x != nil end) |> message_options.__struct__.new()) |> inspect(limit: :infinity) %>
```

Comment:
one more nit, instead of having all this logic in the template can we pass it as a variable like we do the full name? Then this pr is good to go!